apint fails to compile on beta

apint fails to compile (https://crater-reports.s3.amazonaws.com/beta-1.31-1/beta-2018-10-30/reg/apint-0.2.0/log.txt) on beta. Note that the crate must be compiled in test mode.
```
Nov 03 02:59:10.198 INFO kablam!    Compiling apint v0.2.0 (/source)
Nov 03 02:59:13.651 INFO kablam! error[E0597]: borrowed value does not live long enough
Nov 03 02:59:13.651 INFO kablam!    --> src/apint/constructors.rs:429:11
Nov 03 02:59:13.651 INFO kablam!     |
Nov 03 02:59:13.651 INFO kablam! 429 |               .chain([
Nov 03 02:59:13.651 INFO kablam!     |  ____________________^
Nov 03 02:59:13.651 INFO kablam! 430 | |                 u8::max_value(),
Nov 03 02:59:13.651 INFO kablam! 431 | |                 10,
Nov 03 02:59:13.651 INFO kablam! 432 | |                 42,
Nov 03 02:59:13.651 INFO kablam! 433 | |                 99,
Nov 03 02:59:13.651 INFO kablam! 434 | |                 123
Nov 03 02:59:13.651 INFO kablam! 435 | |             ].into_iter()
Nov 03 02:59:13.651 INFO kablam!     | |_____________^ temporary value does not live long enough
Nov 03 02:59:13.651 INFO kablam! 436 |                .map(|v| *v))
Nov 03 02:59:13.651 INFO kablam! 437 |       }
Nov 03 02:59:13.651 INFO kablam!     |       - temporary value only lives until here
Nov 03 02:59:13.651 INFO kablam!     |
Nov 03 02:59:13.651 INFO kablam!     = note: borrowed value must be valid for the static lifetime...
Nov 03 02:59:13.651 INFO kablam! 
Nov 03 02:59:13.657 INFO kablam! error[E0597]: borrowed value does not live long enough
Nov 03 02:59:13.657 INFO kablam!    --> src/apint/constructors.rs:480:11
Nov 03 02:59:13.657 INFO kablam!     |
Nov 03 02:59:13.657 INFO kablam! 480 |               .chain([
Nov 03 02:59:13.657 INFO kablam!     |  ____________________^
Nov 03 02:59:13.657 INFO kablam! 481 | |                 u16::max_value(),
Nov 03 02:59:13.657 INFO kablam! 482 | |                 500,
Nov 03 02:59:13.657 INFO kablam! 483 | |                 1000,
Nov 03 02:59:13.657 INFO kablam! ...   |
Nov 03 02:59:13.657 INFO kablam! 486 | |                 42_000
Nov 03 02:59:13.657 INFO kablam! 487 | |             ].into_iter().map(|v| *v))
Nov 03 02:59:13.657 INFO kablam!     | |_____________^ temporary value does not live long enough
Nov 03 02:59:13.657 INFO kablam! 488 |       }
Nov 03 02:59:13.657 INFO kablam!     |       - temporary value only lives until here
Nov 03 02:59:13.657 INFO kablam!     |
Nov 03 02:59:13.657 INFO kablam!     = note: borrowed value must be valid for the static lifetime...
Nov 03 02:59:13.657 INFO kablam! 
Nov 03 02:59:13.658 INFO kablam! error[E0597]: borrowed value does not live long enough
Nov 03 02:59:13.658 INFO kablam!    --> src/apint/constructors.rs:513:11
Nov 03 02:59:13.658 INFO kablam!     |
Nov 03 02:59:13.658 INFO kablam! 513 |               .chain([
Nov 03 02:59:13.658 INFO kablam!     |  ____________________^
Nov 03 02:59:13.658 INFO kablam! 514 | |                 u32::max_value(),
Nov 03 02:59:13.658 INFO kablam! 515 | |                 1_000_000,
Nov 03 02:59:13.658 INFO kablam! 516 | |                 999_999_999,
Nov 03 02:59:13.658 INFO kablam! 517 | |                 1_234_567_890
Nov 03 02:59:13.658 INFO kablam! 518 | |             ].into_iter().map(|v| *v))
Nov 03 02:59:13.658 INFO kablam!     | |_____________^ temporary value does not live long enough
Nov 03 02:59:13.658 INFO kablam! 519 |       }
Nov 03 02:59:13.658 INFO kablam!     |       - temporary value only lives until here
Nov 03 02:59:13.659 INFO kablam!     |
Nov 03 02:59:13.659 INFO kablam!     = note: borrowed value must be valid for the static lifetime...
Nov 03 02:59:13.659 INFO kablam! 
Nov 03 02:59:13.663 INFO kablam! error[E0597]: borrowed value does not live long enough
Nov 03 02:59:13.663 INFO kablam!    --> src/apint/constructors.rs:544:11
Nov 03 02:59:13.663 INFO kablam!     |
Nov 03 02:59:13.663 INFO kablam! 544 |               .chain([
Nov 03 02:59:13.663 INFO kablam!     |  ____________________^
Nov 03 02:59:13.663 INFO kablam! 545 | |                 u64::max_value(),
Nov 03 02:59:13.663 INFO kablam! 546 | |                 1_000_000_000_000,
Nov 03 02:59:13.663 INFO kablam! 547 | |                 999_999_999_999_999_999,
Nov 03 02:59:13.663 INFO kablam! 548 | |                 0x0123_4567_89AB_CDEF
Nov 03 02:59:13.663 INFO kablam! 549 | |             ].into_iter().map(|v| *v))
Nov 03 02:59:13.663 INFO kablam!     | |_____________^ temporary value does not live long enough
Nov 03 02:59:13.663 INFO kablam! 550 |       }
Nov 03 02:59:13.664 INFO kablam!     |       - temporary value only lives until here
Nov 03 02:59:13.664 INFO kablam!     |
Nov 03 02:59:13.664 INFO kablam!     = note: borrowed value must be valid for the static lifetime...
Nov 03 02:59:13.664 INFO kablam! 
Nov 03 02:59:13.671 INFO kablam! error[E0597]: borrowed value does not live long enough
Nov 03 02:59:13.671 INFO kablam!    --> src/apint/constructors.rs:575:11
Nov 03 02:59:13.671 INFO kablam!     |
Nov 03 02:59:13.671 INFO kablam! 575 |               .chain([
Nov 03 02:59:13.671 INFO kablam!     |  ____________________^
Nov 03 02:59:13.671 INFO kablam! 576 | |                 u128::max_value(),
Nov 03 02:59:13.671 INFO kablam! 577 | |                 1_000_000_000_000_000_000_000_000,
Nov 03 02:59:13.671 INFO kablam! 578 | |                 999_999_999_999_999_999_999_999_999,
Nov 03 02:59:13.671 INFO kablam! 579 | |                 0x0123_4567_89AB_CDEF_FEDC_BA98_7654_3210
Nov 03 02:59:13.671 INFO kablam! 580 | |             ].into_iter().map(|v| *v))
Nov 03 02:59:13.671 INFO kablam!     | |_____________^ temporary value does not live long enough
Nov 03 02:59:13.672 INFO kablam! 581 |       }
Nov 03 02:59:13.672 INFO kablam!     |       - temporary value only lives until here
Nov 03 02:59:13.672 INFO kablam!     |
Nov 03 02:59:13.672 INFO kablam!     = note: borrowed value must be valid for the static lifetime...
Nov 03 02:59:13.672 INFO kablam! 
Nov 03 02:59:14.105 INFO kablam! error: aborting due to 5 previous errors
```

--- a/src/libcore/num/mod.rs
+++ b/src/libcore/num/mod.rs
@@ -2152,6 +2152,7 @@ Basic usage:
 ", $Feature, "assert_eq!(", stringify!($SelfT), "::min_value(), 0);", $EndFeature, "
 ```"),
             #[stable(feature = "rust1", since = "1.0.0")]
+            #[rustc_promotable]
             #[inline]
             pub const fn min_value() -> Self { 0 }
         }
@@ -2168,6 +2169,7 @@ Basic usage:
 stringify!($MaxV), ");", $EndFeature, "
 ```"),
             #[stable(feature = "rust1", since = "1.0.0")]
+            #[rustc_promotable]
             #[inline]
             pub const fn max_value() -> Self { !0 }
         }

--- a/src/test/ui/consts/auxiliary/promotable_const_fn_lib.rs
+++ b/src/test/ui/consts/auxiliary/promotable_const_fn_lib.rs
@@ -1,0 +1,31 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Crate that exports a const fn. Used for testing cross-crate.
+
+#![feature(staged_api, rustc_attrs)]
+#![stable(since="1.0.0", feature = "mep")]
+
+#![crate_type="rlib"]
+
+#[rustc_promotable]
+#[stable(since="1.0.0", feature = "mep")]
+#[inline]
+pub const fn foo() -> usize { 22 }
+
+#[stable(since="1.0.0", feature = "mep")]
+pub struct Foo(usize);
+
+impl Foo {
+    #[stable(since="1.0.0", feature = "mep")]
+    #[inline]
+    #[rustc_promotable]
+    pub const fn foo() -> usize { 22 }
+}

--- a/src/test/ui/consts/promote_fn_calls.rs
+++ b/src/test/ui/consts/promote_fn_calls.rs
@@ -1,0 +1,13 @@
+// compile-pass
+// aux-build:promotable_const_fn_lib.rs
+
+#![feature(nll)]
+
+extern crate promotable_const_fn_lib;
+
+use promotable_const_fn_lib::{foo, Foo};
+
+fn main() {
+    let x: &'static usize = &foo();
+    let x: &'static usize = &Foo::foo();
+}

--- a/src/test/ui/consts/promote_fn_calls_std.rs
+++ b/src/test/ui/consts/promote_fn_calls_std.rs
@@ -1,0 +1,30 @@
+// compile-pass
+
+#![feature(nll)]
+
+fn main() {
+    let x: &'static u8 = &u8::max_value();
+    let x: &'static u16 = &u16::max_value();
+    let x: &'static u32 = &u32::max_value();
+    let x: &'static u64 = &u64::max_value();
+    let x: &'static u128 = &u128::max_value();
+    let x: &'static usize = &usize::max_value();
+    let x: &'static u8 = &u8::min_value();
+    let x: &'static u16 = &u16::min_value();
+    let x: &'static u32 = &u32::min_value();
+    let x: &'static u64 = &u64::min_value();
+    let x: &'static u128 = &u128::min_value();
+    let x: &'static usize = &usize::min_value();
+    let x: &'static i8 = &i8::max_value();
+    let x: &'static i16 = &i16::max_value();
+    let x: &'static i32 = &i32::max_value();
+    let x: &'static i64 = &i64::max_value();
+    let x: &'static i128 = &i128::max_value();
+    let x: &'static isize = &isize::max_value();
+    let x: &'static i8 = &i8::min_value();
+    let x: &'static i16 = &i16::min_value();
+    let x: &'static i32 = &i32::min_value();
+    let x: &'static i64 = &i64::min_value();
+    let x: &'static i128 = &i128::min_value();
+    let x: &'static isize = &isize::min_value();
+}

